src/app/pages/test-results-search/components/advanced-search/advanced-search.ts

```diff
@@ -117,7 +117,8 @@ export class AdvancedSearchComponent {
     if (typeof event.value !== 'string') return;
 
     // Added logic here as it is used on the (ionInput) attribute of the staffNo. input field, sets toggle to unchecked
-    if (event.value === '' && (field === 'staffId')) {
+    if ((event.value === '' && (field === 'staffId') &&
+      (this.selectedTestCentre === undefined || this.selectedTestCentre === null))) {
```

Comment:
Can we simplify this too check i.e.

 `if (event.value === '' && field === 'staffId' && !this.selectedTestCentre) {`